apps/src/dance/ai/AiBlockPreview.tsx

```diff
@@ -1,39 +1,70 @@
 import {BlockSvg, Workspace, WorkspaceSvg} from 'blockly';
-import React, {useEffect, useRef} from 'react';
+import React, {useEffect, useRef, useState} from 'react';
 import moduleStyles from './ai-block-preview.module.scss';
 
 interface AiBlockPreviewProps {
+  fadeIn: boolean;
   generateBlocksFromResult: (workspace: Workspace) => [BlockSvg, BlockSvg];
+  onComplete: () => void;
 }
 
 /**
  * Previews the blocks generated by the AI block in Dance Party.
  */
 const AiBlockPreview: React.FunctionComponent<AiBlockPreviewProps> = ({
+  fadeIn,
   generateBlocksFromResult,
+  onComplete,
 }) => {
   const blockPreviewContainerRef = useRef<HTMLSpanElement>(null);
+  const refTimer = useRef<number | null>(null);
 
+  const [done, setDone] = useState<boolean>(false);
+
+  // Update currentTick using a timer.
   useEffect(() => {
-    const emptyBlockXml = Blockly.utils.xml.textToDom('<xml></xml>');
-    const previewWorkspace: Workspace =
-      Blockly.BlockSpace.createReadOnlyBlockSpace(
-        blockPreviewContainerRef.current,
-        emptyBlockXml,
-        {}
-      );
-
-    const blocksSvg = generateBlocksFromResult(previewWorkspace);
-
-    blocksSvg.forEach((blockSvg: BlockSvg) => {
-      blockSvg.initSvg();
-      blockSvg.render();
-    });
-    Blockly.svgResize(previewWorkspace as WorkspaceSvg);
-  }, [blockPreviewContainerRef, generateBlocksFromResult]);
+    refTimer.current = window.setTimeout(
```

Comment:
Not at all required, but i wonder if this actually might be handy to put into its own reusable custom hook (something like `const done = useDelay(fadeIn ? 5000 : 2500, onComplete)`)